openexr-rs/src/tiledinputfile.rs

```diff
@@ -0,0 +1,970 @@
+                ly,
+            )
+            .into_result()?;
+        }
+
+        Ok(level_height)
+    }
+
+    /// Returns the number of tiles in x direction that cover a level with level
+    /// number `(lx, *)`, where `*` is any number.
+    ///
+    ///	Return value is `(TiledInputFile::level_width(lx) + TiledInputFile::tile_x_size() - 1) / TiledInputFile::tile_x_size()`
+    ///
+    pub fn num_x_tiles(&self, lx: i32) -> Result<i32, Error> {
+        let mut num_x_tiles = 0;
+
+        unsafe {
+            sys::Imf_TiledInputFile_numXTiles(self.inner, &mut num_x_tiles, lx)
+                .into_result()?;
+        }
+
+        Ok(num_x_tiles)
+    }
+
+    /// Returns the number of tiles in y direction that cover a level with level
+    /// number `(*, ly)`, where `*` is any number.
+    ///
+    ///	Return value is `(TiledInputFile::level_height(ly) + TiledInputFile::tile_x_size() - 1) / TiledInputFile::tile_x_size()`
+    ///
+    pub fn num_y_tiles(&self, ly: i32) -> Result<i32, Error> {
+        let mut num_y_tiles = 0;
+
+        unsafe {
+            sys::Imf_TiledInputFile_numYTiles(self.inner, &mut num_y_tiles, ly)
+                .into_result()?;
+        }
+
+        Ok(num_y_tiles)
+    }
+
+    /// returns a 2-dimensional region of valid pixel coordinates for a level
+    /// with level number `(lx, ly)`.
+    ///
+    /// Return value is a Box2i (imath) with the min value
+    /// `(data_window.min.x, data_window.min.y)` and max value
+    /// `(data_window.min.x + TiledInputFile::level_width(lx) - 1, data_window.min.y + TiledInputFile::level_height(ly) - 1)`
+    ///
+    pub fn data_window_for_level<T>(&self, lx: i32, ly: i32) -> Result<T, Error>
+    where
+        T: Bound2<i32>,
+    {
+        let mut data_window =
+            std::mem::MaybeUninit::<sys::Imath_Box2i_t>::uninit();
+
+        unsafe {
+            sys::Imf_TiledInputFile_dataWindowForLevel(
+                self.inner,
+                data_window.as_mut_ptr(),
+                lx,
+                ly,
+            )
+            .into_result()?;
+
+            let data_window: [i32; 4] =
+                std::mem::transmute(data_window.assume_init());
+            Ok(T::from_slice(&data_window))
+        }
+    }
+
+    /// Returns a 2-dimensional region of valid pixel coordinates for a tile
+    /// with tile coordinates `(dx,dy)` and level number `(lx, ly)`.
+    ///
+    /// Return value is a Box2i (imath) with the min value
+    /// `(data_window.min.x + dx * TiledInputFile::tile_x_size(), data_window.min.y + dy * TiledInputFile::tile_y_size())`
+    /// and max value
+    /// `(data_window.min.x + (dx + 1) * TiledInputFile::tile_x_size() - 1, data_window.min.y + (dy + 1) * TiledInputFile::tile_y_size() - 1)`
+    ///
+    pub fn data_window_for_tile<T: Bound2<i32>>(
+        &self,
+        dx: i32,
+        dy: i32,
+        lx: i32,
+        ly: i32,
+    ) -> Result<T, Error> {
+        let mut data_window =
```

Comment:
You don't need to do the whole uninit/transmute dance here. It's cleaner just to do:
```
let mut data_window = [0i32; 4];
// ...
data_window.as_mut_ptr() as *mut sys::Imath_Box2i_t,
// ...
T::from_slice(&data_window)
```

Also, less important but I'm using `B` for Bounds and `V` for Vec (and V2/V3 where necessary) for generic parameters.